test/test-crossbuild.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail
+
+# test if we can build for all OSes and ARCHes.
+
+make crossbuild
```

Comment:
I think having a test for this is awesome, although if we ever break OSX, I might have to add an exit 0 here. Fine with that though.

Reply:
No problem, or just set a GOOSARCHES variable with explicitly supported crossbuilds. As long as i'm arround testing OSX I will do best effort to keep supporting it. If not it should not become a burden on you unless you want to.